File: front/src/services/language/LanguageService.ts
import type { Language, LanguageProficiency, LanguagePurpose, UserLanguage } from './types.ts'

export interface LanguageService {
  languages(activeOnly?: boolean): Promise<Language[]>
  myLanguages(): Promise<UserLanguage[]>
  setUserLanguage(
    languageCode: string,
    purpose: LanguagePurpose,
    proficiency?: LanguageProficiency,
  ): Promise<UserLanguage>
  removeUserLanguage(languageCode: string, purpose: LanguagePurpose): Promise<boolean>
}

File: back/src/main/kotlin/com/jvmvstv_v/back/language/service/LanguageService.kt
package com.jvmvstv_v.back.language.service

import com.jvmvstv_v.back.language.model.Language
import com.jvmvstv_v.back.language.model.LanguageProficiency
import com.jvmvstv_v.back.language.model.LanguagePurpose
import com.jvmvstv_v.back.language.model.UserLanguage

interface LanguageService {
    fun languages(activeOnly: Boolean?): List<Language>
    fun myLanguages(): List<UserLanguage>
    fun setUserLanguage(languageCode: String, purpose: LanguagePurpose, proficiency: LanguageProficiency?): UserLanguage
    fun removeUserLanguage(languageCode: String, purpose: LanguagePurpose): Boolean
}
